ext/dom/xpath.c

```diff
@@ -132,7 +137,7 @@ static void dom_xpath_ext_function_php(xmlXPathParserContextPtr ctxt, int nargs,
 	if (obj->stringval == NULL) {
 		zend_type_error("Handler name must be a string");
 		xmlXPathFreeObject(obj);
-		goto cleanup;
+		goto cleanup_no_callable;
 	}
```

Comment:
You should be able to check that the string is indeed a valid callable before allocating the arguments of it.

Moreover, (for master) you could use the ``named_params`` field instead of allocating each argument individually, should make clean-up easier as you would just need to destroy the HashTable.

Reply:
> You should be able to check that the string is indeed a valid callable before allocating the arguments of it.

Not easy to do, it's a stack machine so the function name comes last. I'd have to store the past arguments anyway so I can't avoid the allocation.

> Moreover, (for master) you could use the named_params field instead of allocating each argument individually, should make clean-up easier as you would just need to destroy the HashTable.

TIL. But that will also be slower because of the hashtable management code.

Reply:
> > You should be able to check that the string is indeed a valid callable before allocating the arguments of it.
> 
> Not easy to do, it's a stack machine so the function name comes last. I'd have to store the past arguments anyway so I can't avoid the allocation.

Okay, that code makes _way_ more sense